src/commons/utils/corio_utils.py

```diff
@@ -265,6 +275,52 @@ def setup_environment():
     os.makedirs(DATA_DIR_PATH, exist_ok=True)
 
 
+def activate_degraded_mode(options: dict) -> None:
+    """Create bucket and degrade node as needed .
+
+    :param options : dictionary to fetch command line params
+    :return : None
+    """
+    bucket = f'object-op-{time.perf_counter_ns()}'.lower()
+    response = S3Bucket(access_key=options.access_key,
+                        secret_key=options.secret_key,
+                        endpoint_url=options.endpoint_url,
+                        use_ssl=options.use_ssl).create_bucket(bucket)
+    if response:
+        os.environ["d_bucket"] = bucket
+
+    if options.degrade_pod:
+        __degrade_nodes(options)
+
+
+def __degrade_nodes(options: dict) -> None:
+    """
+    degrade nodes as needed for execution by shutdown/deleting pod
+    :param options : dictionary to fetch command line params
+    """
+    nodes = CLUSTER_CFG["nodes"]
+    (host, user, password) = (None, None, None)
+    for node in nodes:
+        if node["node_type"] == "master":
+            host = node["hostname"]
+            user = node["username"]
+            password = node["password"]
+            break
+    logical_node = LogicalNode(host=host,
+                               user=user,
+                               password=password)
+    LOGGER.info("Shutdown the data pods safely by making replicas=0")
+    pod_list = logical_node.get_all_pods(pod_prefix=DATA_POD_NAME_PREFIX)
+    for _ in range(literal_eval(os.environ.get("options")).get("degraded_pods")):
+        LOGGER.info("Get pod name to be deleted")
+        pod_name = random.sample(pod_list, 1)[0]
+        pod_list.remove(pod_name)
+        hostname = logical_node.get_pod_hostname(pod_name=pod_name)
+        LOGGER.info("Deleting pod %s", pod_name)
+        logical_node.create_pod_replicas(num_replica=0, pod_name=pod_name)
+        LOGGER.info("Shutdown/deleted pod %s for host %s with replicas=0", pod_name, hostname)
```

Comment:
how are we checking if pod is actually deleted? Also do we want to make any more checks regarding cluster status/services running on this pod and other pods as well, as that will affect IOs?

Reply:
This needs to be taken care in test logic. We may add later in corio  if needed